privacyidea/models.py

```diff
@@ -2127,7 +2135,21 @@ def option_dict(self):
         """
         res = {}
         for option in self.options:
-            res[option.Key] = option.Value
+            if option.Type == "option" or option.Type is None:
+                res[option.Key] = option.Value
+        return res
+
+    @property
+    def header_dict(self):
+        """
+        Return all connected options as a dictionary
+
+        :return: dict
+        """
+        res = {}
+        for header in self.headers:
```

Comment:
You do not need the second relation. 
You can do

~~~~python
for header in self.options:
    if header.Type == "header":
        res[header.key] = header.Value
~~~~